evmap-derive/src/lib.rs

```diff
@@ -0,0 +1,77 @@
+use proc_macro2::{Ident, TokenStream};
+use quote::{quote, quote_spanned};
+use syn::spanned::Spanned;
+use syn::{parse_macro_input, parse_quote, Data, DeriveInput, Fields, GenericParam, Generics, Index};
+
+#[proc_macro_derive(ShallowCopy)]
+pub fn derive_heap_size(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
```

Comment:
This name seems wrong?

Reply:
Lol yes you can tell I based it on the example from `syn`.